### PR TITLE
FLUID-5598: Added "terms" block for defining common string template values for resource loaders

### PR DESCRIPTION
**Pull request by cindyli**
@amb26, can you review this pull request? Thanks.

### CODE REVIEW

File: src/framework/preferences/js/AuxBuilder.js [hunk 1]
```diff
@@ -349,10 +341,17 @@ var fluid_2_0 = fluid_2_0 || {};
...
 
+    // Go over the aux schema to replace terms with their given values.
     fluid.prefs.expandSchema = function (schemaToExpand, indexes, topCommonOptions, elementCommonOptions, mappedDefaults) {
         var auxSchema = fluid.prefs.expandSchemaImpl(schemaToExpand);
         auxSchema.namespace = auxSchema.namespace || "fluid.prefs.created_" + fluid.allocateGuid();
 
+        var terms = fluid.get(auxSchema, "terms");
+        if (terms) {
+            delete auxSchema.terms;
+            fluid.set(auxSchema, ["terms", "terms"], terms);
```
Comment (amb26) [inline src/framework/preferences/js/AuxBuilder.js]: Why does "terms" appear twice here?

Comment (cindyli) [inline src/framework/preferences/js/AuxBuilder.js]: The first "terms" is the name of the component that contains a "terms" option to be pass along. The second "terms" is indeed the "terms" option in this component.

Comment (amb26) [inline src/framework/preferences/js/AuxBuilder.js]: Where can I find the definition of the component named "terms"?

Comment (cindyli) [inline src/framework/preferences/js/AuxBuilder.js]: "terms" is a component constructed and used internally by the builder. 

The basic definition for it - https://github.com/cindyli/infusion/blob/FLUID-5598/src/framework/preferences/js/AuxBuilder.js#L420-L422. The "terms" from the aux schema is passed in as an option.

 Once constructed, this component is used as a grade for constructing prefsEditorLoader (https://github.com/cindyli/infusion/blob/FLUID-5598/src/framework/preferences/js/Builder.js#L122) where its terms option is passed down to loaders.

Comment (amb26) [inline src/framework/preferences/js/AuxBuilder.js]: Thanks cindyli - perhaps we can add a comment in expandSchema explaining to readers what the strategy is and where the site is where this output structure is consumed. Also, this pull request requires a corresponding pull to update https://github.com/fluid-project/infusion-docs/blob/master/src/documents/AuxiliarySchemaForPreferencesFramework.md in infusion-docs

**Comment by cindyli** [comment]
@amb26, this pull request is ready for more reviews. 

The accompanying infusion-docs change - https://github.com/fluid-project/infusion-docs/pull/67

**Pull request merged**